server/src/controllers/user.controllers.ts

```diff
@@ -1,100 +1,113 @@
 import { Request, RequestHandler, Response } from 'express';
-import User from '../models/user.model';
+import { User } from '../models/user.model';
+import { UserMongoType } from '../types/user.types';
 
-const sendResponse = (res: Response, statusCode: number, payload: any) => {
+const sendResponse = (
+  res: Response,
+  statusCode: number,
+  payload: UserMongoType[] | UserMongoType | string
+) => {
   return res.status(statusCode).json(payload);
 };
 
-const createUser: RequestHandler = async (req: Request, res: Response): Promise<any> => {
-  const { ...toCreate } = req.body;
+const createUser: RequestHandler = async (req: Request, res: Response): Promise<void> => {
+  const { ...toCreate }: UserMongoType = req.body;
 
   if (Object.keys(toCreate).length > 0) {
     try {
-      const data = await User.create(toCreate);
+      const data: UserMongoType = await User.create(toCreate);
 
       if (data) {
         sendResponse(res, 200, data);
       } else {
-        sendResponse(res, 404, { error: 'User not created' });
+        sendResponse(res, 404, 'User not created');
       }
     } catch (error: any) {
-      sendResponse(res, 500, { error: error.message });
+      sendResponse(res, 500, error?.message);
```

Comment:
i think we need a type for error handling or just better error handling in general. but don't worry about it now. can be done a bit later, after we finished the basic APIs. just putting this down so hopefully I don't forget